src/wp/FileTransfer.cs

```diff
@@ -210,6 +216,80 @@ public FileTransferProgress(long bTotal = 0, long bLoaded = 0)
         }
 
         /// <summary>
+        /// Helper method to copy all relevant cookies from the WebBrowser control into a header on
+        /// the HttpWebRequest
+        /// </summary>
+        /// <param name="browser">The source browser to copy the cookies from</param>
+        /// <param name="webRequest">The destination HttpWebRequest to add the cookie header to</param>
+        /// <returns>Nothing</returns>
+        private async Task CopyCookiesFromWebBrowser(HttpWebRequest webRequest)
+        {
+            var tcs = new TaskCompletionSource<object>();
+
+            // Accessing WebBrowser needs to happen on the UI thread
+            Deployment.Current.Dispatcher.BeginInvoke(() =>
```

Comment:
I think we can use Invoke() instead of BeginInvoke() here so we don't need to additionally use TaskCompletionSource.